dnd5esheets/front/src/store/index.ts

```diff
@@ -1,491 +1,358 @@
-import { createComputed } from 'solid-js'
-import { createStore, reconcile } from 'solid-js/store'
-import { CharacterSchema } from '~/5esheets-client'
-import { ActionType } from '~/5esheets-client'
+import {
+  Resource,
+  createComputed,
+  createResource,
+  createReaction,
+  createMemo,
+  createEffect,
+  Accessor,
+} from 'solid-js'
+import {
+  SetStoreFunction,
+  Store,
+  createStore,
+  produce,
+  reconcile,
+  unwrap,
+} from 'solid-js/store'
+import { createRouteData } from 'solid-start'
+import { CharacterSchema, ListCharacterSchema } from '~/5esheets-client'
 import { Proficiency } from '~/5esheets-client'
 import { Scores } from '~/5esheets-client'
 import { SaveProficiencies } from '~/5esheets-client'
 import { SkillProficiencies } from '~/5esheets-client'
+import { CharacterService } from '~/5esheets-client'
+import { applyEffects, computeEffect, Effect } from '~/effects'
+import { AssignmentOperator } from '~/effects/parser'
 
-export const cycleProficiency = (proficiency: number) => (proficiency + 1) % 3
+type Join<K, P> = K extends string | number
+  ? P extends string | number
+    ? `${K}${'' extends P ? '' : '.'}${P}`
+    : never
+  : never
+
+type Split<S extends string, D extends string> = string extends S
+  ? string[]
+  : S extends ''
+  ? []
+  : S extends `${infer T}${D}${infer U}`
+  ? [T, ...Split<U, D>]
+  : [S]
+
+type Prev = [
+  never,
+  0,
+  1,
+  2,
+  3,
+  4,
+  5,
+  6,
+  7,
+  8,
+  9,
+  10,
+  11,
+  12,
+  13,
+  14,
+  15,
+  16,
+  17,
+  18,
+  19,
+  20,
+  ...0[],
+]
 
-type ScoreKey = keyof Scores
-type SaveProficiencyKey = keyof SaveProficiencies
-type SkillProficiencyKey = keyof SkillProficiencies
+type Leaves<T, D extends number = 10> = [D] extends [never]
```

Comment:
I'm not sure what `Leaves` refers to

Reply:
I've tried to put a comment there: https://github.com/brouberol/5esheets/pull/200/files#diff-61178c0abe6fbdba6e7be2fcbc02e55ffb0939f433470a284f735730a8716da7R58
Basically, it's quite advanced typing to turn an object type (e.g. `{a: {b: string, c: number}}`) into the type of the paths that can access the leaves of this object (i.e. `'a.b' | 'a.c'`).
The goal is to type which target are possible to use in effects, but it might prove too hard to implement and not useful enough.